### PR TITLE
Part4

### PR DESCRIPTION
**Pull request by MasonSelf**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -274,10 +179,17 @@ struct FloatType
...
         {
             return *value;
         }
+
+        FloatType& pow( float ft );
+        FloatType& pow( const IntType& it );
+        FloatType& pow( const FloatType& ft );
+        FloatType& pow( const DoubleType& dt );
+
+        FloatType& powInternal( float ft );
```
Comment (matkatmusic) [inline main.cpp:192]: `powInternal` should be private.   that's why it's `internal`
- [ ] revise

File: main.cpp [hunk 2]
```diff
@@ -294,10 +206,17 @@ struct DoubleType
...
         {
             return *value;
         }
+
+        DoubleType& pow( double dt );
+        DoubleType& pow( const IntType& it );
+        DoubleType& pow( const FloatType& ft );
+        DoubleType& pow( const DoubleType& dt );
+
+        DoubleType& powInternal( double dt );
```
Comment (matkatmusic) [inline main.cpp:219]: `powInternal` should be private.   that's why it's `internal`
- [ ] revise

File: main.cpp [hunk 3]
```diff
@@ -314,10 +233,17 @@ struct IntType
...
         {
             return *value;
         }
+
+        IntType& pow( int it );
+        IntType& pow( const IntType& it );
+        IntType& pow( const FloatType& ft );
+        IntType& pow( const DoubleType& dt );
+
+        IntType& powInternal( int it );
```
Comment (matkatmusic) [inline main.cpp:246]: `powInternal` should be private.   that's why it's `internal`
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
one minor change needed

File: main.cpp (hunk 1 above)
Comment (MasonSelf) [reply main.cpp:192]: Roger that

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**